GVFS/GVFS.Service/RepoRegistry.cs

```diff
@@ -232,31 +202,38 @@ public void AutoMountRepos(int sessionId)
                                 RepoRegistration registration = RepoRegistration.FromJson(entry);
 
                                 string errorMessage;
-                                string normalizedEnlistmentRootPath = registration.EnlistmentRoot;
-                                if (GVFSPlatform.Instance.FileSystem.TryGetNormalizedPath(registration.EnlistmentRoot, out normalizedEnlistmentRootPath, out errorMessage))
+                                string enlistmentPath = registration.EnlistmentRoot;
+
+                                // Try and normalize the enlistment path if the volume is available, otherwise just take the
+                                // path verbatim.
+                                string volumePath = GVFSPlatform.Instance.FileSystem.GetVolumeRoot(registration.EnlistmentRoot);
+                                if (GVFSPlatform.Instance.FileSystem.IsVolumeAvailable(volumePath))
                                 {
-                                    if (!normalizedEnlistmentRootPath.Equals(registration.EnlistmentRoot, StringComparison.OrdinalIgnoreCase))
+                                    string normalizedPath;
+                                    if (GVFSPlatform.Instance.FileSystem.TryGetNormalizedPath(registration.EnlistmentRoot, out normalizedPath, out errorMessage))

```

Comment:
`TryGetNormalizedPath` will fail if the path doesn't actually exist. For repositories on volumes which are still locked by BitLocker this means we can read the registry entries! Not sure why we'd ever get non-normalised paths in the registry, but changed this to only bother trying if the volume is actually available.